### ISSUE TITLE
Deptry could lookup metedata only once

### ISSUE THREAD

**Issue by akeeman** [fpgmaas/deptry#336]
<continuing discussion started [in #333](https://github.com/fpgmaas/deptry/pull/333#issuecomment-1512486795)>

`deptry.dependency.Dependency::found` and `deptry.dependency.Dependency::find_metadata` seem to be unused outside of the class itself. They may be internalized, so the class can be simplified by skipping creating `found` all together.

```python
try:
  top_levels += self._get_top_level_module_names_from_top_level_txt()
  if not top_levels:
      top_levels += self._get_top_level_module_names_from_record_file()
except PackageNotFoundError:
    pass   # may do `top_levels.append(name.replace("-", "_").lower())` here, see #335
```

**Comment by fpgmaas** [comment]
To add, from [here](https://github.com/fpgmaas/deptry/pull/333#discussion_r1169986424):

Currently, `deptry` contains some potentially confusing behavior. When `python-dotenv` is in the dependencies, but not installed (so it has no metadata), with the following configuration:

```
[tool.deptry.package_module_name_map]
python-dotenv = [
    "dotenv",
]
```

a warning is still shown:

```log
Warning: Package 'python-dotenv' not found in current environment. Assuming its corresponding module name is 'python_dotenv'.
```

I think this is now undesired behavior? As a user I would expect that, since I manually specified which module I want the dependency to map to, I do not expect to see a warning, since I already specified how I want this package to be dealt with.
